node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -601,74 +599,19 @@ impl MainReactor {
-                            self.chainspec.deploy_config.max_ttl,
-                        )? {
-                            return Ok(Some(SyncBackInstruction::TtlSynced));
-                        }
-                    }
+                if let Some(synched) = self.synched(&highest_orphaned_block_header)? {
+                    return Ok(Some(synched));
                 }
+                let (sync_hash, sync_era) =
+                    self.sync_hash_and_era(&highest_orphaned_block_header)?;
+                debug!(?sync_era, %sync_hash, "KeepUp: historical sync target era and block hash");
 
-                let parent_hash = highest_orphaned_block_header.parent_hash();
-                debug!(?highest_orphaned_block_header, %parent_hash, "KeepUp: highest orphaned historical block");
-                match self.storage.read_block_header(parent_hash) {
-                    Ok(Some(parent_block_header)) => {
-                        // even if we don't have a complete block (all parts and dependencies)
-                        // we may have the parent's block header; if we do we also
-                        // know its era which allows us to know if we have the validator
-                        // set for that era or not
-                        debug!(
-                            ?parent_block_header,
-                            "KeepUp: found parent block header for historical block in storage"
-                        );
-                        Ok(Some(SyncBackInstruction::Sync {
-                            parent_hash: parent_block_header.block_hash(),
-                            era_id: parent_block_header.era_id(),
-                        }))
-                    }
-                    Ok(None) => {
-                        debug!(%parent_hash, "KeepUp: did not find historical block header in storage");
-                        let era_id = match highest_orphaned_block_header.era_id().predecessor() {
-                            None => EraId::from(0),
-                            Some(predecessor) => {
-                                // we do not have the parent header and thus don't know what era
-                                // the parent block is in (it could be the same era or the previous
-                                // era). we assume the worst case and ask for the earlier era's
-                                // proof; subtracting 1 here is safe
-                                // since the case where era id is 0 is
-                                // handled above
-                                predecessor
-                            }
-                        };
-                        Ok(Some(SyncBackInstruction::Sync {
-                            parent_hash: *parent_hash,
-                            era_id,
-                        }))
-                    }
-                    Err(err) => Err(err.to_string()),
-                }
+                self.validator_matrix
```

Comment:
nit: it would be good to preserve the previous comment here

Reply:
it is retained, it was extracted into the synched function below along w/ the logic